File: ViewAnimation-OC/ViewAnimation-OC/ViewController.m
//
//  ViewController.m
//  ViewAnimation-OC
//
//  Created by watchman on 2017/7/2.
//  Copyright © 2017年 Hervey. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (nonatomic,strong)UIButton *loginButton;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.loginButton = [[UIButton alloc]initWithFrame:CGRectMake(-394, 230, self.view.frame.size.width- 20*2, 50)];
    self.loginButton.backgroundColor = [UIColor colorWithRed:50/255.0 green:185/255.0 blue:170/255.0 alpha:1.0];
    [self.loginButton setTitle:@"登录" forState:UIControlStateNormal];
    [self.view addSubview:self.loginButton];
}

- (void)viewWillAppear:(BOOL)animated {
//    [UIView animateWithDuration:1.0 animations:^{
//        self.loginButton.frame = CGRectMake(20, self.loginButton.frame.origin.y, self.loginButton.frame.size.width, self.loginButton.frame.size.height);
//    }];
    [UIView beginAnimations:nil context:nil];
    [UIView setAnimationDuration:1.0];
    self.loginButton.frame = CGRectMake(20, self.loginButton.frame.origin.y, self.loginButton.frame.size.width, self.loginButton.frame.size.height);
    [UIView commitAnimations];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: ViewAnimation-Swift/ViewAnimation/ViewController.swift
//
//  ViewController.swift
//  ViewAnimation
//
//  Created by watchman on 2017/7/2.
//  Copyright © 2017年 Hervey. All rights reserved.
//

import UIKit
var loginButtion:UIButton?
class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        loginButtion = UIButton(frame:CGRect(x:-394,y:230,width:self.view.frame.width-20*2,height:50))
        loginButtion!.backgroundColor = UIColor(red:50/255.0,green:185/255.0,blue:170/255.0,alpha:1.0)
        loginButtion!.setTitle("登录", for:UIControlState.normal)
        self.view.addSubview(loginButtion!)
        
    }
    override func viewWillAppear(_ animated: Bool) {
//        UIView.animate(withDuration: 1,animations:{
//            loginButtion!.frame = CGRect(x:20,y:loginButtion!.frame.origin.y,width:loginButtion!.frame.width,height:loginButtion!.frame.height)
//        })
        
        UIView.beginAnimations(nil, context: nil)//动画开始
        UIView.setAnimationDuration(1);//动画周期设置
         loginButtion!.frame = CGRect(x:20,y:loginButtion!.frame.origin.y,width:loginButtion!.frame.width,height:loginButtion!.frame.height) //动画位置设置
        UIView.commitAnimations() //动画提交
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

